### ISSUE TITLE
Is this Project abandoned?

### ISSUE THREAD

**Issue by kaykaul** [cjb/GitTorrent#75]
Is this project abandoned? I was searching for real distributed peer to peer git systems and found this awesome repo. Kudos for creating this technology. What is the roadmap for this project?

**Comment by cjb** [comment]
Hi! Not really abandoned, but I haven't been writing new code. Some things we could use help with:
- an Electron desktop client
- integrating a source of decentralized single-namespace naming, like blockstack, but in JS
- working out how to distribute/decentralize pull requests and issues

**Comment by eonist** [comment]
What stopped the progress? What part was to time consuming? Complexity or did you discover an essential part that wasn't easy to implement? Great effort anyway!

**Comment by adlai** [comment]
@eonist haven't you ever heard of the [larval stage](http://www.catb.org/jargon/html/L/larval-stage.html)?

Have patience, and the muse will strike again!

**Comment by eonist** [comment]
A bit ambiguous. Waiting for the muse then.

**Comment by cjb** [comment]
The usual things happened: new job, and new kid.  :)  I'm also hoping for an easy to use blockstack clone in JS to show up without me having to write it.

**Comment by eonist** [comment]
@cjb: Thanks for your response. And congrats on the new chapters in your life! What do you think about the saying that we shouldn't try to fit everything into the blockchain? That the blockchain may not be suitable for everything etc? And would there be an alternative to this for this project? Or any ideas surrounding this dilemma?

**Comment by cjb** [comment]
Hm.  Are you talking about bitcoin's blockchain in particular, or the concept of shared public decentralized append-only data structures?

I think decentralized apps should use shared public decentralized append-only data structures because it's very difficult to form a secure consensus without them, and there are many problems that require reaching consensus.

I don't have any particular attachment to Bitcoin.  It's the coin people are most likely to own, which makes it the most useful at the moment.  But if (e.g. Ethereum) an altcoin explicitly endorses decentralized naming (I think Ethereum does) and is practical -- i.e. people can get it, and it's well-designed, and it doesn't introduce too much friction to the app -- then that altcoin could totally be used here as well.

**Comment by eonist** [comment]
@cjb: Right, Bitcoin powers the blockchain as it currently stands. Something has to finance the life of such a database, thats bitcoin at the moment. Isn't there a limit to how much data the block chain could hold, i mean if everyone started to store things in the blockchain it would get pretty crowded. Also Wont there be a cost to storing things in the blockchain?

**Comment by cjb** [comment]
Arguably the cost is already paid for by the OP_RETURN fee, and using OP_RETURN shows other users of Bitcoin that this is a null transaction that they don't have to store forever.  (In technical terms, OP_RETURN doesn't create "UTXO entries".)  And we're talking about small strings, not storing entire files.

So my sense is that this is about the least objectionable use of Bitcoin to store data.

**Comment by eonist** [comment]
Right. So you need: blockstack-cli or blockstack-server in js? And what exactly are you storing in the blockchain? Authentication or? Also Issues and pull-requests can be done at a later stage maybe and just instead focusing on making git working.

**Comment by cjb** [comment]
Storing strings that look like:

```
@gittorrent!cjb!81e24205d4bac8496d3e13282c90ead5045f09ea

(@service!username!pubkey)
```

**Comment by cjb** [comment]
Maybe just blockstack-cli is fine.  (But it'd be ideal if the whole stack was JS.)

**Comment by kaykaul** [comment]
Hmm It seems someone has started the JS library.
https://github.com/blockstack/blockstack-js

**Comment by ghost** [comment]
WRT the blockchain bits, I've opened #79 to discuss switching the concept to Ethereum instead.

**Comment by inoas** [comment]
A very valuable part of github is its commenting, review system etc, if not its most valuable part.
What about an alternate route of running an app server decentralized.
The work to do would be to create an environment where you could run decentralized apps. Then you could leverage gitlab or any other similar tool and run it on top... in theory.

alternative to that you would need to make features similar to github in a mobile/desktop client that just pushes data to the network ... so there is no need for a server - really hard if you ask me but would be awesome.

good luck!

**Comment by ghost** [comment]
@inoas What about [ipfs](https://github.com/ipfs/ipfs) ?

**Comment by eonist** [comment]
@asciiascetic ipfs-> Distributed static content  vs https://github.com/amark/gun Distributed dynamic websites. Also keep in mind that ipfs is coupled with FILECOIN as MaidSafe(.)net is coupled to: SAFECOIN

**Comment by frankenstein91** [comment]
does it still work?

**Comment by rektide** [comment]
Hey sad day on the net as Microsoft immediately quickly caves the fuck under to these long-persisting turd-sandwiches & their cheap loser users-have-no-rights humanity-hating scum-shit.

https://github.com/github/dmca/blob/master/2020/10/2020-10-23-RIAA.md

This project was one of my first thoughts. We need this kind of defensive tech very badly.

**Comment by itspngu** [comment]
bringitback

**Comment by eonist** [comment]
Someone should do a business analysis on the feasibility of this. Map it out so to speak. Then get investors on board. Maybe even apply for Y-combinator. This would be disruptive!

**Comment by ghost** [comment]
That's what this US-centric corporate hellscape scenario needs to inject a bit of liberty, more US-centric corporates! 🙃

And later if Gittorrent, Inc was successful maybe Microsoft could buy it and drop all the good stuff from the DHT 😬

Really though, all the good stuff in life is made by people who care, not people who want a payday. If Gittorrent is to have its day, people need to:

A) Use it
B) Contribute to it
(Optional) C) Port it
(Optional) D) Run volunteer, geographically/jurisdictionally diverse seedboxes to keep repos online

**Comment by frankenstein91** [comment]
> Hey sad day on the net as Microsoft immediately quickly caves the fuck under to these long-persisting turd-sandwiches & their cheap loser users-have-no-rights humanity-hating scum-shit.
> 
> https://github.com/github/dmca/blob/master/2020/10/2020-10-23-RIAA.md
> 
> This project was one of my first thoughts. We need this kind of defensive tech very badly.

Thank you for informing us that the youtube-dl project has been deleted. I have informed the German press about this and hope there will be a reaction for a popular tool.

**Comment by eonist** [comment]
> That's what this US-centric corporate hellscape scenario needs to inject a bit of liberty, more US-centric corporates! 🙃 And later if Gittorrent, Inc was successful maybe Microsoft could buy it and drop all the good stuff from the DHT 😬 Really though, all the good stuff in life is made by people who care, not people who want a payday. If Gittorrent is to have its day, people need to: A) Use it B) Contribute to it (Optional) C) Port it (Optional) D) Run volunteer, geographically/jurisdictionally diverse seedboxes to keep repos online

I think this and other Great projects Were progress has frozen to ice because the core-dev suddenly had to focus on survival, aka work for big corp, would benefit greatly from VC money. The world isn’t black and white. Lots of fantastic VCs out there. Not just evil ones. I’m VC backed and I make a free decentralised password manager, which hopefully one day will make the world a better place. The OSS model for projects like this will and has never worked. It’s too big of a task. IMO

**Comment by theAkito** [comment]
>the youtube-dl project has been deleted.

Oh my god. I can't believe this. I am speechless. They keep taking away the best stuff, over and over. I'm shocked.

Well, good to know that I'm NOT crazy in mirroring quite a few projects, in fear of them disappearing from one day to another...

**Comment by Giszmo** [comment]
People in the nostr community are thinking about how to have a decentralized GitHub, ideally with issue tracking and code reviews and everything but as a bare minimum something that allows you to `git remote add something gittorrent:<hash of first commit or something cryptographic>`.

A shame this project just fizzled out :(
